app/views/categories/_categories.html.haml

```diff
@@ -0,0 +1,18 @@
+%ul.category-summary
+  - categories.each do |category|
+    - if category.has_children?
+      - count = category.children.inject(0) do |memo, child|
```

Comment:
Can you use [`Enumerable#sum`](http://api.rubyonrails.org/classes/Enumerable.html#method-i-sum) to simplify calculation?
``` ruby
category.children.map {|child| child.posts.published.count }.sum